File: sentimental-credit/credit.py
cardNumber = input("Card number:")
intCardNumber = int(cardNumber)
sum = 0
mult = 0
rawValue = 0


for i in range(len(cardNumber)):
    print(f"{intCardNumber}")
    # Sum digits
    is_sum = i % 2 == 0
    if (is_sum):
        sum = sum + (intCardNumber % 10)
    # Multiply then sum digts
    else:
        multResult = 2 * (intCardNumber % 10)
        if multResult > 9:
            mult = multResult - 9
        else:
            mult = multResult

    rawValue = (mult + sum) + rawValue
    mult = 0
    sum = 0

    if intCardNumber < 100 and intCardNumber > 9:
        firstTwoDigits = intCardNumber

    intCardNumber = intCardNumber // 10


cardValue = ((rawValue) % 10)

if cardValue != 0:
    print(f"INVALID")
elif (len(cardNumber) > 12 and len(cardNumber) < 17 ) and (firstTwoDigits // 10 == 4):
    print("VISA")
elif (len(cardNumber) == 15 and (firstTwoDigits == 34 or firstTwoDigits ==37)):
    print("AMEX")
elif (len(cardNumber) == 16 and (firstTwoDigits > 50 and firstTwoDigits < 56)):
    print("MASTERCARD")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)

{
    long CardNumber;

    // get credit card number
    CardNumber = get_long("Credit card number: ");

    int sumOrMult = 0;
    int multiplySum = 0;
    int naoTaUsando = 0;
    int digitCount = 0;
    int firstTwoDigits = 0;

    // runs for each digit validation and necessary actions
    for (int i = 10; CardNumber >= 1; CardNumber = CardNumber / 10)
    {
        digitCount++;
        if (CardNumber < 100 && firstTwoDigits == 0)
        {
            firstTwoDigits = ((CardNumber % 100));
        }
        if (sumOrMult == 0)
        {
            multiplySum = multiplySum + (CardNumber % 10);
            sumOrMult = 1;
        }
        else
        {
            if ((CardNumber % 10) * 2 >= 10)
            {
                multiplySum = multiplySum + ((CardNumber % 10) * 2 - 9);
            }
            else
            {
                multiplySum = multiplySum + (CardNumber % 10) * 2;
            }
            sumOrMult = 0;
        }
    }

    // print results based on values
    if (multiplySum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        if (digitCount >= 13 && digitCount <= 16 && (firstTwoDigits / 10 == 4))
        {
            printf("VISA\n");
        }
        else if (digitCount == 15 && (firstTwoDigits == 34 || firstTwoDigits == 37))
        {
            printf("AMEX\n");
        }
        else if (digitCount == 16 && (firstTwoDigits >= 51 && firstTwoDigits <= 55))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
}
